File: Objective-C/Patterns/MVC/DumbColorsView.m
#import "MVC.h"
#import "ColorsController.h"
#import "DumbColorsView.h"

@implementation DumbColorsView

+ viewWithModel:(Model *)aModel andName:(const char *)aName
{
    return OOB_AUTORELEASE([[self alloc] initWithModel:aModel andName:aName]);
}

- initWithModel:(Model *)aModel andName:(const char *)aName
{
    [self initWithModel:aModel];
    if (aName != NULL) {
        name = OOB_RETAIN([NSString stringWithCString:aName]);
    } else {
        name = @"none";
    }
    return self;
}

- (Controller *)makeController
{
    return [ColorsController controllerWithView:self];
}

- (void)draw
{
    ++drawCount;
}

- (unsigned long)drawCount
{
    return drawCount;
}

- (void)dealloc
{
    OOB_RELEASE(name);
    [super dealloc];
}

@end

File: C++/Patterns/MVC/DumbColorsView.cpp
#include "MVC.hpp"
#include "ColorsController.hpp"

namespace oobench {

    class DumbColorsView : public View {
        std::string name;
        unsigned long drawCount;
    public:
        DumbColorsView(Model& theModel, char* theName = "none") 
                : View(theModel), name(theName) {
        }

        DumbColorsView(const DumbColorsView& anotherDumbColorsView) 
                : View(anotherDumbColorsView) {
            name = anotherDumbColorsView.name;
        }

        virtual ~DumbColorsView() {
        }

        Controller* makeController() {
            return new ColorsController(*this);
        }

        ColorsController& getColorsController() const {
            return (ColorsController&)getController();
        }

        virtual void draw() {
            ++drawCount;
        }

         unsigned long getDrawCount() const {
             return drawCount;
         }

    };
}
